numpy/core/src/multiarray/arrayfunction_override.c

```diff
@@ -329,6 +323,72 @@ array_implement_array_function(
 }
 
 
+/*
+ * Implements the __array_function__ protocol for a Python function, as described in
+ * in NEP-18. See numpy.core.overrides for a full docstring.
+ */
+NPY_NO_EXPORT PyObject *
+array_implement_array_function(
+    PyObject *NPY_UNUSED(dummy), PyObject *positional_args)
+{
+    PyObject *implementation, *public_api, *relevant_args, *args, *kwargs;
+
+    if (!PyArg_UnpackTuple(
+            positional_args, "implement_array_function", 5, 5,
+            &implementation, &public_api, &relevant_args, &args, &kwargs)) {
+        return NULL;
+    }
+
+    /* Remove `like=` kwarg, which is NumPy-exclusive and thus not present
+     * in downstream libraries.
+     */
+    PyObject *like_key = PyUnicode_FromString("like");
+    if (PyDict_CheckExact(kwargs) && PyDict_Contains(kwargs, like_key)) {
+        PyDict_DelItem(kwargs, like_key);
+    }
+    Py_DECREF(like_key);
+
+    return array_implement_array_function_internal(
+        implementation, public_api, relevant_args, args, kwargs);
+}
+
+
+/*
+ * Implements the __array_function__ protocol for a C function. Added as an extension
+ * to NEP-18 in an effort to bring NEP-35 to life with minimal dispatch overhead.
+ */
+NPY_NO_EXPORT PyObject *
+array_implement_c_array_function(
+    const char *function_name, PyObject *args, PyObject *kwargs)
+{
+    if (kwargs == NULL)
+        return NULL;
+
+    PyObject *relevant_args;
+
+    /* Remove `like=` kwarg, which is NumPy-exclusive and thus not present
+     * in downstream libraries. If that key isn't present, return NULL and
+     * let originating call to continue.
+     */
+    PyObject *like_key = PyUnicode_FromString("like");
```

Comment:
This would be worth caching with other strings at startup.

Reply:
Good idea, I'll do that.

Reply:
Use an interned string rather than a regular one: https://docs.python.org/3/c-api/unicode.html#c.PyUnicode_InternFromString

Reply:
The other ones are in `np_ma_...` variables I think.

Reply:
Done.